src/main/java/org/broadinstitute/consent/http/service/DatasetService.java

```diff
@@ -57,6 +58,36 @@ public DataSet getDatasetWithPropertiesById(Integer datasetId) {
         return dataset;
     }
 
+    public Optional<DataSet> updateDataset(DataSetDTO dataset, Integer datasetId, Integer userId) {
+        Timestamp now = new Timestamp(new Date().getTime());
+
+        DataSet old = getDatasetWithPropertiesById(datasetId);
+        Set<DataSetProperty> oldProperties = old.getProperties();
+
+        List<DataSetPropertyDTO> dtos = dataset.getProperties();
+        List<DataSetProperty> jsonProperties = processDatasetProperties(datasetId, dtos);
+
+        List<DataSetProperty> propertiesToAdd = jsonProperties.stream()
+              .filter(p -> oldProperties.stream()
+                    .noneMatch(op -> op.getPropertyKey() == p.getPropertyKey()))
+              .collect(Collectors.toList());
+
+        List<DataSetProperty> propertiesToUpdate = jsonProperties.stream()
+              .filter(p -> oldProperties.stream()
+                    .noneMatch(op -> p.equals(op)))
+              .collect(Collectors.toList());
```

Comment:
Do you need to also look at properties to delete? If I `PUT` a dataset with fewer props than the one that's there, I would expect that the missing props would be removed. This is in contrast to a `PATCH` where you're only making updates to the specific things that were passed in.